internal/service/streamprocessor/data_source_schema.go

```diff
@@ -10,42 +10,49 @@ import (
-			},
-			"state": schema.StringAttribute{
-				Computed:            true,
-				Description:         "The state of the stream processor.",
-				MarkdownDescription: "The state of the stream processor.",
-			},
-			"stats": schema.StringAttribute{
-				Computed:            true,
-				Description:         "The stats associated with the stream processor.",
-				MarkdownDescription: "The stats associated with the stream processor.",
-			},
+		Attributes: DSAttributes(true),
+	}
+}
+
+func DSAttributes(withArguments bool) map[string]schema.Attribute {
+	return map[string]schema.Attribute{
+		"id": schema.StringAttribute{
+			Computed:            true,
+			Description:         "Unique 24-hexadecimal character string that identifies the stream processor.",
+			MarkdownDescription: "Unique 24-hexadecimal character string that identifies the stream processor.",
+		},
+		"instance_name": schema.StringAttribute{
+			Required:            withArguments,
+			Computed:            !withArguments,
+			Description:         "Human-readable label that identifies the stream instance.",
+			MarkdownDescription: "Human-readable label that identifies the stream instance.",
+		},
+		"pipeline": schema.StringAttribute{
+			Computed:            true,
+			Description:         "Stream aggregation pipeline you want to apply to your streaming data.",
+			MarkdownDescription: "Stream aggregation pipeline you want to apply to your streaming data.",
+		},
+		"processor_name": schema.StringAttribute{
+			Required:            withArguments,
+			Computed:            !withArguments,
+			Description:         "Human-readable label that identifies the stream processor.",
+			MarkdownDescription: "Human-readable label that identifies the stream processor.",
+		},
+		"project_id": schema.StringAttribute{
+			Required:            withArguments,
+			Computed:            !withArguments,
+			Description:         "Unique 24-hexadecimal digit string that identifies your project. Use the [/groups](#tag/Projects/operation/listProjects) endpoint to retrieve all projects to which the authenticated user has access.\n\n**NOTE**: Groups and projects are synonymous terms. Your group id is the same as your project id. For existing groups, your group/project id remains the same. The resource and corresponding endpoints use the term groups.",
+			MarkdownDescription: "Unique 24-hexadecimal digit string that identifies your project. Use the [/groups](#tag/Projects/operation/listProjects) endpoint to retrieve all projects to which the authenticated user has access.\n\n**NOTE**: Groups and projects are synonymous terms. Your group id is the same as your project id. For existing groups, your group/project id remains the same. The resource and corresponding endpoints use the term groups.",
```

Comment:
Should id be uppercase ID?

Reply:
We are using lowercase throughout the API and Terraform provider docs; therefore, I'll keep it that way